src/pages/Dashboard.tsx

```diff
@@ -1,35 +1,188 @@
+            }
+        })();
+    }, [])
+
+    // Generate and store hardcoded insight object for now
+    const createNewInsight = async () => {
+        const hardcodedInsight: Insight = {
+            createdBy: 'userId',
+            createDate: 1,
+            lastUpdated: 2,
+            ticketId: 'DATASCI-2864:',
+            title: 'FireBase - Title blah blah blah',
+            imgs: [{src: Logo, alt: 'google image'}],
+            outcomeNumber: 21,
+            keyStat: '21% increase per annum',
+            keyNumber: '21%',
+            tldr: 'string string string string string string string string string',
+            takeaways: 'string string string string string string string string',
+            requests: ['RequestId1'],
+            collaborators: ['UserId1']
+        };
+        try {
+            const insightId: string = await addInsight(TEST_ORGANIZATION, TEST_GROUP, hardcodedInsight);
+            displayedInsights[insightId] = hardcodedInsight;
+            setDisplayedInsights({...displayedInsights});
+        } catch (error) {
+            // Replace with user visible messaging
+            console.log("Could not create new insight");
+        }
+    }
+
+    // Generate and store hardcoded request object for now
+    const createNewFiledRequest = async () => {
+        const hardcodedFiledRequest: FiledRequest = {
+            createdBy: "BigBossId",
+            createDate: 123,
+            lastUpdated: 123,
+            title: "FireBase - Title blah blah blah",
+            assignee: "CodeMonkeyId",
+            priority: "High",
+            status: "Assigned",
+            deadline: 123
+        };
+        try {
+            const filedRequestId: string = await addFiledRequest(TEST_ORGANIZATION, TEST_GROUP, hardcodedFiledRequest);
+            displayedRequests[filedRequestId] = hardcodedFiledRequest;
+            // TODO: Refactor, current state is a bit hacky to force React to update
+            setDisplayedRequests({...displayedRequests});
+        } catch (error) {
+            // Replace with user visible messaging
+            console.log("Could not create new request");
+        }
+    }
+
+    const generateDeleteFiledRequestHandler = (filedRequestId: string): Function => {
+        return async () => {
+            try {
+                await deleteInsight(TEST_ORGANIZATION, TEST_GROUP, filedRequestId);
+                delete displayedRequests[filedRequestId];
+                // TODO: Refactor, current state is a bit hacky to force React to update
+                setDisplayedRequests({...displayedRequests});
+            } catch {
+                console.log("Could not delete filed request");
+            }
+        }
+    };
+
+    const generateDeleteInsightHandler = (insightId: string): Function => {
+        return async () => {
+            try {
+                await deleteInsight(TEST_ORGANIZATION, TEST_GROUP, insightId);
+                delete displayedInsights[insightId];
```

Comment:
don't mutate displayedInsights directly, make a copy first then delete the index and set the copy to the state, also what does generate refer to?

Reply:
Done